packages/uicore/src/components/DateInput/DateInput.tsx

```diff
@@ -236,7 +235,7 @@ export const DateInput: React.FC<DateInputProps> = props => {
         type="text"
         value={state.value}
         intent={state.intent}
-        placeholder={formatDateTime}
+        placeholder={placeholder}
```

Comment:
NIT: We could have kept both (if it makes sense).
defaultTo(placeholder, formatDateTime);